scala-thoughts.md

```diff
@@ -0,0 +1,5 @@
+I find the balancd between object-oriented programming and functional programming is a strong point about Scala. It is fairly straightforward to create a new object in Scala just as it is in other object-oriented programming languages. Within definitions of objects, we can take advantage of the functional side of Scala. Based on my previous experience with functional programming languages, it was nice to be able to have and use higher-order functions. Pattern matching is also something I personally enjoyed using in Scala. Pattern matching is a strong point about Haskell, another functional programming language, that seems to be just as useful in Scala (although I don't know if there are any differences yet).

```

Comment:
Agreed, it's really nice that Scala provides a platform for the paradigm of our choice and the tools available in each of those disciplines!